rust/src/rune_coral.rs

```diff
@@ -1,64 +1,44 @@
 use crate::{ffi, Error, InferenceContext, TensorDescriptor};
 use std::{
-    ffi::{CString, OsStr},
+    ffi::{CString},
     mem::MaybeUninit,
-    ptr::{self, NonNull},
-    sync::Arc,
+    ptr::{self, NonNull}
 };
 
-/// A safe wrapper around `librunecoral` which has been dynamically loaded at
-/// runtime.
-pub struct RuneCoral {
-    inner: Arc<ffi::RuneCoral>,
-}
-
-impl RuneCoral {
-    pub fn load(path: impl AsRef<OsStr>) -> Result<Self, libloading::Error> {
-        unsafe {
-            let inner = ffi::RuneCoral::new(path)?;
-            Ok(RuneCoral {
-                inner: Arc::new(inner),
-            })
-        }
-    }
+pub fn create_inference_context(
```

Comment:
Now we aren't going through the `RuneCoral` object, can we make this a static method on the `InferenceContext` type? Maybe call it `InferenceContext::create()` or something.